modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java

```diff
@@ -339,17 +343,27 @@ public void testGetMetaOutboundConnector() throws IOException {
     @Test
     public void testAddIdPWithUserDefinedAuthenticator() throws IOException {
 
+        String baseIdentifier = "federatedAuthenticators.authenticators.find { it.authenticatorId == '" +
+                FEDERATED_AUTHENTICATOR_ID + "' }.";
+
         String body = idpCreatePayload.replace(FEDERATED_AUTHENTICATOR_ID_PLACEHOLDER,
                 userDefinedAuthenticatorPayload.getAuthenticatorId());
         body = body.replace(FEDERATED_AUTHENTICATOR_PLACEHOLDER,
                 userDefinedAuthenticatorPayload.convertToJasonPayload());
         body = body.replace(IDP_NAME_PLACEHOLDER, CUSTOM_IDP_NAME);
+
         Response response = getResponseOfPost(IDP_API_BASE_PATH, body);
         response.then()
                 .log().ifValidationFails()
                 .assertThat()
                 .statusCode(HttpStatus.SC_CREATED)
-                .header(HttpHeaders.LOCATION, notNullValue());
+                .header(HttpHeaders.LOCATION, notNullValue())
+                .body("name", equalTo(CUSTOM_IDP_NAME))
+                .body(baseIdentifier + "authenticatorId", equalTo(FEDERATED_AUTHENTICATOR_ID))
+                .body(baseIdentifier + "name", equalTo(new String(Base64.getDecoder().decode(FEDERATED_AUTHENTICATOR_ID))))
+                .body(baseIdentifier + "isEnabled", equalTo(true))
+                .body(baseIdentifier + "definedBy", equalTo(DEFINED_BY_USER))
+                .body(baseIdentifier + "tags", Matchers.hasItems("Custom"));
 
```

Comment:
Assert self, check other places as well.

Reply:
This is updated with e73225fc2e8068e574a4280d5206f44f5aaf2ceb